src/utils.cpp

```diff
@@ -285,6 +286,98 @@ std::string get_pid_exe(pid_t pid)
   return get_pid_exe(std::to_string(pid));
 }
 
+std::string get_proc_maps(const std::string &pid)
+{
+  std::error_code ec;
+  std_filesystem::path proc_path{ "/proc" };
+  proc_path /= pid;
+  proc_path /= "maps";
+
+  if (!std_filesystem::exists(proc_path, ec))
+    return "";
+
+  return proc_path.string();
+}
+
+std::string get_proc_maps(pid_t pid)
+{
+  return get_proc_maps(std::to_string(pid));
+}
+
+std::set<std::string> get_mapped_file_paths_for_pid(pid_t pid)
```

Comment:
I don't see a declaration for this -- maybe mark static or put into anon namespace?